tournamentcontrol/competition/tests/test_thumbnails.py

```diff
@@ -0,0 +1,179 @@
+"""
+Tests for YouTube thumbnail management functionality.
+"""
+
+import io
+from unittest.mock import Mock, patch
+
+from django.test import TestCase
+
+from tournamentcontrol.competition._mediaupload import MediaDatabaseUpload
+from tournamentcontrol.competition.models import Season, Match
+from tournamentcontrol.competition.tests.factories import SeasonFactory, MatchFactory
+
+
+class MediaDatabaseUploadTestCase(TestCase):
+    """Test the MediaDatabaseUpload class."""
+    
+    def setUp(self):
+        self.image_data = b'\x89PNG\r\n\x1a\n\x00\x00\x00\rIHDR\x00\x00\x00\x01\x00\x00\x00\x01\x08\x06\x00\x00\x00\x1f\x15\xc4\x89\x00\x00\x00\rIDATx\x9cc\x00\x01\x00\x00\x05\x00\x01\r\n-\xdb\x00\x00\x00\x00IEND\xaeB`\x82'
+        self.mimetype = 'image/png'
+    
+    def test_media_database_upload_creation(self):
+        """Test creating MediaDatabaseUpload with binary data."""
+        upload = MediaDatabaseUpload(self.image_data, self.mimetype)
+        
+        self.assertEqual(upload.mimetype(), self.mimetype)
+        self.assertEqual(upload.size(), len(self.image_data))
+        self.assertFalse(upload.resumable())
+        self.assertTrue(upload.has_stream())
+    
+    def test_media_database_upload_resumable(self):
+        """Test creating resumable MediaDatabaseUpload."""
+        upload = MediaDatabaseUpload(self.image_data, self.mimetype, resumable=True)
+        self.assertTrue(upload.resumable())
+    
+    def test_getbytes(self):
+        """Test getting bytes from upload."""
+        upload = MediaDatabaseUpload(self.image_data, self.mimetype)
+        
+        # Get first 10 bytes
+        chunk = upload.getbytes(0, 10)
+        self.assertEqual(chunk, self.image_data[:10])
+        
+        # Get middle chunk
+        chunk = upload.getbytes(5, 15)
+        self.assertEqual(chunk, self.image_data[5:15])
+    
+    def test_from_bytes_with_data(self):
+        """Test creating MediaDatabaseUpload from bytes."""
+        upload = MediaDatabaseUpload.from_bytes(self.image_data, self.mimetype)
+        
+        self.assertIsNotNone(upload)
+        self.assertEqual(upload.mimetype(), self.mimetype)
+        self.assertEqual(upload.size(), len(self.image_data))
+    
+    def test_from_bytes_no_data(self):
+        """Test creating MediaDatabaseUpload with no image data."""
+        upload = MediaDatabaseUpload.from_bytes(None, self.mimetype)
+        
+        self.assertIsNone(upload)
+    
+    def test_from_bytes_no_mimetype(self):
+        """Test error when image data exists but no MIME type."""
```

Comment:
We can determine the MIME type by introspecting the bytes, so we should never need to throw.